### PR TITLE
[W11H03] Fix many issues in W11H03

### PR DESCRIPTION
**Pull request by BjarneHa**
This pull-request fixes the following issues:
- `getContacts()` not being allowed to return an empty map
- `testMessageHuge()` output being different on Windows
- `testMessage()` not accepting an arbitrarily long message buffer
- Not providing reasons for many assertions (Added Zulip links)

### CODE REVIEW

**Review by nilsreichardt — APPROVED** [review]
LGTM

I'm waiting for a second review by @felixhauptmann before merging

File: w11h03/test/pgdp/networking/SocketTest.java [hunk 1]
```diff
@@ -283,36 +283,36 @@ void testMessages() throws IOException, InterruptedException {
...
 
     @Test
     @Order(9)
     @DisplayName("sendMessage() – huge")
     void testMessagesHuge() throws IOException, InterruptedException {
         // Ideally, we would use resources here, but that seems difficult with the current tests setup
-        String lipsum = Files.readString(Path.of("./test/pgdp/networking/lipsum.txt"));
-        String lipsumTruncated = Files.readString(Path.of("./test/pgdp/networking/lipsum_truncated.txt"));
+        String lipsum = Files.readString(Path.of("./test/pgdp/networking/lipsum.txt")).replace("\r", "");
+        String lipsumTruncated = Files.readString(Path.of("./test/pgdp/networking/lipsum_truncated.txt")).replace("\r", "");
```
Comment (felixhauptmann) [inline w11h03/test/pgdp/networking/SocketTest.java:315]: There are platforms using only '\r' as EOL. This change adds support for these platforms.

```suggestion
        String lipsumTruncated = Files.readString(Path.of("./test/pgdp/networking/lipsum_truncated.txt")).replaceAll("\\r\\n?", "\n");
```

File: w11h03/test/pgdp/networking/SocketTest.java [hunk 2]
```diff
@@ -283,36 +283,36 @@ void testMessages() throws IOException, InterruptedException {
...
     }
 
     @Test
     @Order(9)
     @DisplayName("sendMessage() – huge")
     void testMessagesHuge() throws IOException, InterruptedException {
         // Ideally, we would use resources here, but that seems difficult with the current tests setup
-        String lipsum = Files.readString(Path.of("./test/pgdp/networking/lipsum.txt"));
-        String lipsumTruncated = Files.readString(Path.of("./test/pgdp/networking/lipsum_truncated.txt"));
+        String lipsum = Files.readString(Path.of("./test/pgdp/networking/lipsum.txt")).replace("\r", "");
```
Comment (felixhauptmann) [inline w11h03/test/pgdp/networking/SocketTest.java:314]: There are platforms using only '\r' as EOL. This change adds support for these platforms.

```suggestion
        String lipsum = Files.readString(Path.of("./test/pgdp/networking/lipsum.txt")).replaceAll("\\r\\n?", "\n");
```

**Review by felixhauptmann — CHANGES_REQUESTED** [review]
Other than that LGTM

**Pull request merged**